### PR TITLE
Delegation tab for the view topic page.

### PR DESCRIPTION
**Pull request by andorp**

### CODE REVIEW

**Comment by fisx** [comment]
It looks like the `DelegateeLists` confuses delegates and delegatees somewhere.  Running with the last commit and looking at a topics page, all delegations look are reflexive, and there can be many of them:

```
[(AUID {_unAUID = 92},[AUID {_unAUID = 92},AUID {_unAUID = 92},AUID {_unAUID = 92},AUID {_unAUID = 92}]),(AUID {_unAUID = 147},[AUID {_unAUID = 147},AUID {_unAUID = 147}]),(AUID {_unAUID = 138},[AUID {_unAUID = 138},AUID {_unAUID = 138}]),(AUID {_unAUID = 129},[AUID {_unAUID = 129},AUID {_unAUID = 129}]),(AUID {_unAUID = 122},[AUID {_unAUID = 122},AUID {_unAUID = 122}]),(AUID {_unAUID = 112},[AUID {_unAUID = 112},AUID {_unAUID = 112}]),(AUID {_unAUID = 110},[AUID {_unAUID = 110},AUID {_unAUID = 110}]),(AUID {_unAUID = 108},[AUID {_unAUID = 108},AUID {_unAUID = 108}]),(AUID {_unAUID = 107},[AUID {_unAUID = 107},AUID {_unAUID = 107}]),(AUID {_unAUID = 98},[AUID {_unAUID = 98},AUID {_unAUID = 98}]),(AUID {_unAUID = 94},[AUID {_unAUID = 94},AUID {_unAUID = 94}]),(AUID {_unAUID = 90},[AUID {_unAUID = 90},AUID {_unAUID = 90}]),(AUID {_unAUID = 74},[AUID {_unAUID = 74},AUID {_unAUID = 74}]),(AUID {_unAUID = 73},[AUID {_unAUID = 73},AUID {_unAUID = 73}]),(AUID {_unAUID = 56},[AUID {_unAUID = 56},AUID {_unAUID = 56}]),(AUID {_unAUID = 54},[AUID {_unAUID = 54},AUID {_unAUID = 54}]),(AUID {_unAUID = 46},[AUID {_unAUID = 46},AUID {_unAUID = 46}]),(AUID {_unAUID = 40},[AUID {_unAUID = 40},AUID {_unAUID = 40}]),(AUID {_unAUID = 33},[AUID {_unAUID = 33},AUID {_unAUID = 33}]),(AUID {_unAUID = 145},[AUID {_unAUID = 145}]),(AUID {_unAUID = 144},[AUID {_unAUID = 144}]),(AUID {_unAUID = 143},[AUID {_unAUID = 143}]),(AUID {_unAUID = 142},[AUID {_unAUID = 142}]),(AUID {_unAUID = 135},[AUID {_unAUID = 135}]),(AUID {_unAUID = 133},[AUID {_unAUID = 133}]),(AUID {_unAUID = 132},[AUID {_unAUID = 132}]),(AUID {_unAUID = 123},[AUID {_unAUID = 123}]),(AUID {_unAUID = 120},[AUID {_unAUID = 120}]),(AUID {_unAUID = 118},[AUID {_unAUID = 118}]),(AUID {_unAUID = 117},[AUID {_unAUID = 117}]),(AUID {_unAUID = 114},[AUID {_unAUID = 114}]),(AUID {_unAUID = 111},[AUID {_unAUID = 111}]),(AUID {_unAUID = 104},[AUID {_unAUID = 104}]),(AUID {_unAUID = 103},[AUID {_unAUID = 103}]),(AUID {_unAUID = 102},[AUID {_unAUID = 102}]),(AUID {_unAUID = 100},[AUID {_unAUID = 100}]),(AUID {_unAUID = 95},[AUID {_unAUID = 95}]),(AUID {_unAUID = 93},[AUID {_unAUID = 93}]),(AUID {_unAUID = 91},[AUID {_unAUID = 91}]),(AUID {_unAUID = 88},[AUID {_unAUID = 88}]),(AUID {_unAUID = 87},[AUID {_unAUID = 87}]),(AUID {_unAUID = 85},[AUID {_unAUID = 85}]),(AUID {_unAUID = 82},[AUID {_unAUID = 82}]),(AUID {_unAUID = 79},[AUID {_unAUID = 79}]),(AUID {_unAUID = 76},[AUID {_unAUID = 76}]),(AUID {_unAUID = 71},[AUID {_unAUID = 71}]),(AUID {_unAUID = 69},[AUID {_unAUID = 69}]),(AUID {_unAUID = 68},[AUID {_unAUID = 68}]),(AUID {_unAUID = 65},[AUID {_unAUID = 65}]),(AUID {_unAUID = 64},[AUID {_unAUID = 64}]),(AUID {_unAUID = 61},[AUID {_unAUID = 61}]),(AUID {_unAUID = 60},[AUID {_unAUID = 60}]),(AUID {_unAUID = 57},[AUID {_unAUID = 57}]),(AUID {_unAUID = 55},[AUID {_unAUID = 55}]),(AUID {_unAUID = 53},[AUID {_unAUID = 53}]),(AUID {_unAUID = 52},[AUID {_unAUID = 52}]),(AUID {_unAUID = 47},[AUID {_unAUID = 47}]),(AUID {_unAUID = 44},[AUID {_unAUID = 44}]),(AUID {_unAUID = 41},[AUID {_unAUID = 41}]),(AUID {_unAUID = 39},[AUID {_unAUID = 39}]),(AUID {_unAUID = 26},[AUID {_unAUID = 26}]),(AUID {_unAUID = 24},[AUID {_unAUID = 24}]),(AUID {_unAUID = 21},[AUID {_unAUID = 21}])]
```

**Comment by fisx** [comment]
opened #682.  (we can do that separately.)

**Comment by fisx** [comment]
@andorp could you take another look?  please revert or squash the last commit, i just pushed that so it's easier for you to debug.

thanks!

File: src/Persistent/Pure.hs [hunk 1]
```diff
@@ -642,10 +642,10 @@ allDelegationScopes = do
             <> (DScopeIdeaId  . view _Id <$> ideas)
 
 scopeDelegatees :: AUID User -> DScope -> EQuery [Delegation]
-scopeDelegatees uid scope =
-    Delegation scope uid
+scopeDelegatees delegate scope =
+    (\delegatee -> Delegation scope delegatee delegate)
```
Comment (andorp) [inline src/Persistent/Pure.hs]: The fix.

File: src/Persistent/Idiom.hs [hunk 2]
```diff
@@ -228,7 +228,7 @@ newtype DelegateeLists = DelegateeLists [(User, [User])]
 delegateeLists :: Bool -> [(User, [User])] -> DelegateeLists
 delegateeLists omitEmpty = DelegateeLists . s . f
   where
-    s = reverse . sortBy (compare `on` (length . snd))
+    s = sortBy (flip compare `on` (length . snd))
```
Comment (fisx) [inline src/Persistent/Idiom.hs]: hlint is smart!  (-:

**Comment by fisx** [comment]
the rebase didn't make reviewing any easier.  did you change anything else worth noting besides the above-mentioned fix?

anyway, thanks for that fix.  i'm going to assume everything is fine now.  (-:

**Comment by andorp** [comment]
Nope.

**Pull request merged**